source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -248,15 +248,17 @@ Network::IoResult SslSocket::doWrite(Buffer::Instance& write_buffer, bool end_st
   while (bytes_to_write > 0) {
     // TODO(mattklein123): As it relates to our fairness efforts, we might want to limit the number
     // of iterations of this loop, either by pure iterations, bytes written, etc.
+    const auto slice = write_buffer.maybeLinearize(16384, 4096);
```

Comment:
Some of the concerns about resume from a different region could be addressed by skipping the call to linearize when doing a retry.  When retrying, we know that the first slice contains roughly bytes_to_retry_